### PR TITLE
MAGN-6541 With Automatically selected there is no user feedback when re-evaluating intensive geometry

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

This PR implements a spinner (in background) when computing a long graph.
  - UI thread will be blocked for seconds in Rendering stage.
  - Status message will be displayed right after the Spinner. 
  - Spinner runs in the background - both Automatic and RUN click.

spinner

  - NodeModel has a new property IsRendered. Initially IsRendered will be set to false during every Evaluation phase. As the nodes are rendered, this property will be set to true. I used this property to compute the background spinner. Instead of using a counter, this property will allow me to collapse the background spinner. In addition to that, I can get the spinner for watch nodes as well. 

### Declarations

Check these if you believe they are true

- [x ] The code base is in a better state after this PR
- [] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers
@ikeough

### CODE REVIEW

**Comment by ramramps** [comment]
@ikeough  - Fixed the merge issue with  RunTypeEnabledConverter.

**Comment by ikeough** [comment]
@ramramps A few issues:

- Repo needs to be merged with master. I tried this locally, but continued to experience issues with `RunSettingsEnabledConverter`.
- Spinner shows up on top of the Start page if the workflow is set to Automatic running.
- Spinner doesn't disappear when rendering is complete. The attached image shows the spinner and the status message still visible after rendering is complete.
image
- Changing the size of Dynamo does not properly re-layout the status message. The status message gets stuck out in space.
image
- The spinner should be next to the RunSettings control, but there is a large space before the spinner and the status message.
image

**Comment by ramramps** [comment]
@ikeough  - Fixed the Merge conflict. Addressed the comments. Please take a look.

File: src/DynamoCoreWpf/Properties/Resources.resx [hunk 1]
```diff
@@ -1882,6 +1888,7 @@ Do you want to install the latest Dynamo update?</value>
     <value>Move the selected path upward</value>
     <comment>Tool-tip for up arrow</comment>
   </data>
+>>>>>>> e5327fdf2e677efe295052526068eab37b44a4c6
```
Comment (pboyer) [inline src/DynamoCoreWpf/Properties/Resources.resx]: Merge conflict mistakes?

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -1841,7 +1844,10 @@ Do you want to install the latest Dynamo update?</value>
   </data>
   <data name="AddButton" xml:space="preserve">
     <value>Add</value>
-  </data>
+  </data> 
+  <data name="TesselationCompletedMessage" xml:space="preserve">
```
Comment (pboyer) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: `Tessellation` spelling error

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 3]
```diff
@@ -1586,14 +1586,17 @@ Do you want to install the latest Dynamo update?</value>
   <data name="RunTypeToolTipPeriodicallyEnabled" xml:space="preserve">
     <value>Run at the specified interval.</value>
   </data>
-  <data name="RunCompletedMessage" xml:space="preserve">
-    <value>Run completed.</value>
+  <data name="EvalCompletedMessage" xml:space="preserve">
+    <value>Tessalating....</value>
```
Comment (pboyer) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: Tessellating... spelling error.

File: src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs [hunk 4]
```diff
@@ -228,6 +260,8 @@ public RunSettingsViewModel(RunSettings settings, HomeWorkspaceViewModel workspa
             this.workspaceViewModel = workspaceViewModel;
             this.dynamoViewModel = dynamoViewModel;
 
+            //this.dynamoViewModel.PropertyChanged +=dynamoViewModel_PropertyChanged;
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs]: Delete dead code.

File: src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs [hunk 5]
```diff
@@ -216,6 +216,38 @@ public Visibility RunButtonVisibility
...
+        public string ShowRunMessage
+        {
+            get
+            {
+                return showRunMessage;
+                //dynamoViewModel.ShowRunMessage;
+            }
+            set
+            {
+                //dynamoViewModel.ShowRunMessage = value;
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs]: Delete commented out code...

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 6]
```diff
@@ -390,14 +391,40 @@ public HelixWatch3DViewModel BackgroundPreviewViewModel
...
+
+        //public string ShowRunMessage
+        //{
+        //    get { return showRunMessage; }
+        //    set
+        //    {
+        //        showRunMessage = value;
+        //        RaisePropertyChanged("ShowRunMessage");
+        //    }
+        //}
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Delete dead code.

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 7]
```diff
@@ -1882,6 +1888,7 @@ Do you want to install the latest Dynamo update?</value>
     <value>Move the selected path upward</value>
     <comment>Tool-tip for up arrow</comment>
   </data>
+>>>>>>> e5327fdf2e677efe295052526068eab37b44a4c6
```
Comment (pboyer) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: Merge conflict mistakes...

**Comment by ikeough** [comment]
@Racel We need to design a location for the progress indicator, now that we will support multiple home workspaces and the bottom bar will be gone(potentially).

**Comment by Racel** [comment]
@ikeough - I worked on this yesterday, but forgot to post to YT. I will post a proposal to MAGN-8288...

**Comment by Racel** [comment]
@ikeough @ramramps - I updated MAGN-8288. Please see the new mockups. I am proposing a progress icon in the tab itself. This is similar to any web browser. If possible, it might be nice to only show the icon when something is in progress or an error has occurred. There is no need to show the 'ready' state if everything is good and nothing is being evaluated/rendered in the tab.

**Comment by pboyer** [comment]
@Racel Did you consider using the node itself as the progress indicator?

**Comment by Racel** [comment]
@pboyer- i had not considered this. But, can you elaborate on what you are thinking? I am worried that having a progress indicator on every node or having the progress indicator moving from node to node might be too busy/distracting....If you have any comments on the specific designs, please put them in MAGN-8288. It makes it easier for me to track everyone's feedback :)

**Comment by pboyer** [comment]
@Racel floodeditor.com has something like this

flood

**Comment by Racel** [comment]
Thanks, @pboyer. For multiple workspaces, will users be able to edit the graph while a node is evaluating? Or is there a UI Freeze? I could see where seeing the node evaluation state could be valuable, but I think we should try to address this in a separate epic/user story as this is not necessarily a 'home workspace' issue. Does that sound ok? Let me know, and I will file a separate story for it.

**Pull request closed without merging**